documentation/ci-fix.md

```diff
@@ -21,11 +21,21 @@ Or if you want to fix specified service, use the complete path, not relative:
 
 ```
 npm install
-npm run prettier -- --write "<path to repo>/azure-rest-api-specs/specification/**/*.json"
+npx prettier --write "<path to repo clone>/azure-rest-api-specs/specification/**/*.json"
```

Comment:
FYI the need for this to be `npx` is explained here: https://prettier.io/docs/en/cli.html

Reply:
@mikeharder any thoughts on the best way to run prettier? Should we just use npx all the time and avoid having the scripts in the project.json? Also if you use npx does it read the version from package.json or does it always run the latest version? I'm still trying to sort out the usages best options when to use npx or not.

Reply:
One aspect is to install `prettier` (and its transitive deps) from a lockfile, to guarantee repeatable builds.  I think this should already be covered by the `package-lock.json` file in the repo root.

I'm pretty sure `npx` will use the version of `prettier` (and transitive deps) installed from the lockfile, but it would be good to test this to verify.

Reply:
If you run `npx prettier` in the repo before running `npm install`, it tries to install latest, so it appears this is ignoring the `package-lock.json` in the root:

```
~/azure-rest-api-specs$ git clean -xdf
Removing node_modules/

~/azure-rest-api-specs$ npx prettier --version
Need to install the following packages:
  prettier@2.8.8
Ok to proceed? (y) n
```

However, if you run `npm install` first, then `npx prettier` uses the version installed from the lockfile:

```
~/azure-rest-api-specs$ npm install
added 247 packages, and audited 248 packages in 4s

~/azure-rest-api-specs$ npm ls | grep prettier
├── @types/prettier@2.7.2
├── prettier@2.8.4

~/azure-rest-api-specs$ npx prettier --version
2.8.4
```

Reply:
I think our best option will be to tell users to run a script like `npm run prettier-fix`, this way we can modify the `prettier-fix` script in `package.json` to do whatever we want, including running `tsc` (ideally on only the needed directories) before running `prettier`.  If we tell users to directly call `npx prettier`, we have no way to modify the behavior or run pre/post steps.

Reply:
Thanks @mikeharder this is exactly the type of advise I was looking for. I agree we should avoid telling folks to use npx as it isn't necessarily consistent with what we have in our package.json and package.lock.json file. I think npm install and npm run prettier-fix is the best option.

Reply:
One challenge with npm scripts is if you need to pass different arguments.  For example, the current `prettier-fix` script calls `prettier --write specification/**/*.json`.  So if you wanted to only run this command on a subdirectory, you'd need to either use `npx prettier` from the command-line, or we could add another script named just `prettier` so you can use `npm run prettier -- <your-arguments-here>`:

```
  "scripts": {
    "preprettier": "tsc",
    "prettier": "prettier",
```

Reply:
I think they will want to scope the run so having the prettier script should be fine. We could also update the prettier-fix to not pass the full path but allow that to be passed as a parameter and then have a perttier-fix-all or something like that to cover all if needed.